File: api/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use SoftDeletes;
    protected $fillable = [
        'name', 'sku', 'product_type_id', 'collection_id',
        'description', 'base_price', 'unit_cost', 'is_active',
    ];

    protected $casts = [
        'base_price' => 'decimal:2',
        'unit_cost' => 'decimal:2',
        'is_active' => 'boolean',
    ];

    protected $appends = ['total_stock'];

    public function getTotalStockAttribute(): int
    {
        return (int) ($this->stocks_sum_quantity ?? 0);
    }

    public function productType(): BelongsTo
    {
        return $this->belongsTo(ProductType::class, 'product_type_id');
    }

    public function collection(): BelongsTo
    {
        return $this->belongsTo(Collection::class, 'collection_id');
    }

    public function stocks(): HasMany
    {
        return $this->hasMany(Stock::class);
    }

    public function orderItems(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    /** Colores pre-configurados para este producto */
    public function colors(): BelongsToMany
    {
        return $this->belongsToMany(Color::class, 'product_color')
                    ->withPivot('sort_order')
                    ->orderByPivot('sort_order');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Inventory/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using AdminHiitop.Api.Domain.Catalog.Entities;
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Inventory.Entities;

public sealed class Product : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Sku { get; set; }
    public int? ProductTypeId { get; set; }
    public int? CollectionId { get; set; }
    [NotMapped]
    public int? UnitMeasureId { get; set; }
    public string? Description { get; set; }
    public decimal BasePrice { get; set; }
    public decimal UnitCost { get; set; }
    public bool IsActive { get; set; } = true;

    public ProductType? ProductType { get; set; }
    public Collection? Collection { get; set; }
    [NotMapped]
    public UnitMeasure? UnitMeasure { get; set; }
    public ICollection<Stock> Stocks { get; set; } = new List<Stock>();
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<ProductColor> ProductColors { get; set; } = new List<ProductColor>();
    public ICollection<PromotionItem> PromotionItems { get; set; } = new List<PromotionItem>();
}
